review/fileToReview.js

```diff
@@ -1,73 +1,90 @@
+ * @param {string} authorization Authorization header to authenticate the user
  * @param {jws} jws Jws token, mandatory if we get one from login request
  * @param {Number} id Account id
- * @param {Number} page Number of the page
- * @param {Object} previousTransactions Previous page of transactions (To ckeck for dupes)
+ * @param {Number} page Current page number
+ * @param {Object} previousTransactions Previous page of transactions (To check for duplicates)
  * @return {Object} All transactions available on the page
  */
-async function fetchTransactions(fromDate, authorization, jws = null, id, page, previousTransactions) {
-	console.log(`--- Fetch Trasactions page n°${page} ---`);
-	try {
-    var headers = {"Authorisation":  authorization }
+const fetchTransactions = async (
+  fromDate,
+  authorization,
+  jws = null,
+  id,
+  page,
+  previousTransactions
+) => {
+  try {
+    let headers = {
+      authorization,
+      "Content-type": "application/json",
+      Accept: "application/json",
+    };
 
     if (jws) {
       headers = {
-        "Authorisation": authorization,
-        "jws": jws,
-        "Content-type": "application/json",
-        "Accept": "application/json"
-      }
-    } else {
-      headers = {
-        "Authorisation": authorization,
-        "Content-type": "application/json",
-        "Accept": "application/json",
-      }
+        ...headers,
+        jws,
+      };
     }
 
-	  var {code, response } = await doRequest('GET',
-      domain + '/accounts/'+ id + '/transactions?' + `page=${page}`,
-      headers);
+    const { code, response } = await doRequest(
+      "GET",
+      `${domain}/accounts/${id}/transactions?page=${page}`,
+      headers
+    );
+
+    if (validateResponse(response) && code === 200) {
+      if (response.data.meta && response.data.meta.hasPageSuivante) {
+        const mouvements = response.data.Mouvements;
+        if (!mouvements) return [];
 
+        const date = mouvements[mouvements.length - 1].dateValeur;
+        if (date <= fromDate) return [];
 
-		if (response && code == 200 && response.data) {
-      if (response.data.meta) {
-        if (response.data.meta.hasPageSuivante) {
-          let mouvements = response.data.Mouvements;
-          var date = mouvements[mouvements.length -1].dateValeur;
-          if (date <= fromDate) {
-            console.log("FromDate is Reached - we don't need more transaction");
-          } else {
-            // if we have mouvements
-            if (mouvements) {
-              if (assertTransactions(mouvements)) {
-                return [];
-              } else {
-                console.log(`Push transactions from page ${page}`);
-              }
-            } else {
-              throw new Error("Empty list of transactions ! " + JSON.stringify(previousTransactions));
-            }
-            let nextPagesTransactions = fetchTransactions(fromDate, authorization, (jws || null), id, page + 1, mouvements);
```

Comment:
Mostly a point of view but I wouldn't use recursion, I would use basic while loop and break when finished.